### PR TITLE
Update ts, eslint, prettier

### PR DESCRIPTION
**Pull request by rin-st**
## Description

- updated typescript and eslint/prettier related deps
- removed `@typescript-eslint/eslint-plugin` from nextjs `package.json` since it's already included to `eslint-config-next`
- explicit `"plugins": ["@trivago/prettier-plugin-sort-imports"]` to avoid warnings. From [docs](https://github.com/trivago/prettier-plugin-sort-imports?tab=readme-ov-file#usage) 
- added `check-types` script for hardhat
- a little bit formatting/fix types and disabled eslint for a few places after updates

Note: `eslint` v9 is still [not supported](https://github.com/vercel/next.js/issues/64409) by nextjs, so I used  last v8 version

To test take code snippet [here](https://github.com/scaffold-eth/scaffold-eth-2/pull/905#issue-2457103242)

## Additional Information

- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

### CODE REVIEW

File: packages/nextjs/package.json [hunk 1]
```diff
@@ -37,21 +37,20 @@
...
+    "eslint-config-next": "^14.2.15",
+    "eslint-config-prettier": "^8.10.0",
+    "eslint-plugin-prettier": "^5.2.1",
     "postcss": "^8.4.45",
-    "prettier": "^2.8.4",
+    "prettier": "^3.3.3",
     "tailwindcss": "^3.4.11",
     "type-fest": "^4.6.0",
-    "typescript": "5.5.3",
+    "typescript": "5.6.3",
```
[the remark this answers is not in the document]
Comment (rin-st) [reply packages/nextjs/package.json]: Thanks, updated [f7cb86d](https://github.com/scaffold-eth/scaffold-eth-2/pull/963/commits/f7cb86da8d555ff090a659cf497b86ac6b938954)

File: packages/hardhat/package.json [hunk 2]
```diff
@@ -25,26 +26,26 @@
...
     "hardhat-deploy": "^0.12.4",
     "hardhat-deploy-ethers": "^0.4.2",
     "hardhat-gas-reporter": "^2.2.1",
-    "prettier": "^2.8.4",
+    "prettier": "^3.3.3",
     "solidity-coverage": "^0.8.13",
     "ts-node": "^10.9.1",
     "typechain": "^8.1.0",
-    "typescript": "^5.1.6"
+    "typescript": "^5.6.3"
```
[the remark this answers is not in the document]
Comment (rin-st) [reply packages/hardhat/package.json]: [f7cb86d](https://github.com/scaffold-eth/scaffold-eth-2/pull/963/commits/f7cb86da8d555ff090a659cf497b86ac6b938954)

**Comment by rin-st** [comment]
Thanks for the great comments!

> Also maybe its good chance to solve this warnings in .eslintrc files ?

[7c82a33](https://github.com/scaffold-eth/scaffold-eth-2/pull/963/commits/7c82a330f12a8a4f3425530104b5ed8a08a155e0)

> Also in tailwind.config.js I see an error but on main it doesn't seem to appear:

It turns out that by default `next lint` checks only [some folders](https://nextjs.org/docs/app/api-reference/cli/next#next-lint-options) but not the full nextjs dir. I changed it [9599964](https://github.com/scaffold-eth/scaffold-eth-2/pull/963/commits/95999648a2f0cbd47c9e0ad0cef2137ad079277c), so all the nextjs folder is linted now, including `tailwind.config.js`

Plus:
- Removed parser which is already included into eslint-config-next ([parser](https://github.com/vercel/next.js/blob/canary/packages/eslint-config-next/index.js#L97)) and eslint [rule](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/recommended.ts#L31) included to typescript-eslint:recommended (which again included to `eslint-config-next`)
- I met https://github.com/scaffold-eth/scaffold-eth-2/issues/901 again. I changed prettier file to `.prettierrc.js` and [changed import](https://github.com/trivago/prettier-plugin-sort-imports/issues/131#issuecomment-1230926769), because of `@trivago/prettier-plugin-sort-imports` causes bug with underlines in vscode. It's a pity that I couldn't find it earlier, maybe https://github.com/scaffold-eth/scaffold-eth-2/issues/901 wont even started

**Comment by technophile-04** [comment]
> It turns out that by default next lint checks only [some folders](https://nextjs.org/docs/app/api-reference/cli/next#next-lint-options) but not the full nextjs dir. I changed it [9599964](https://github.com/scaffold-eth/scaffold-eth-2/pull/963/commits/95999648a2f0cbd47c9e0ad0cef2137ad079277c), so all the nextjs folder is linted now, including tailwind.config.js

Umm I didn't get this changes 😅, also maybe its bad to pass it whole dir? and we should keep its as it was earlier what `next lint` by defults lints? Since next team might have researched to put all necessary to lint and ignore unnecessary stuff? 

Also I see `  // eslint-disable-next-line @typescript-eslint/no-require-imports` to tailwind config so that it doesn't complain anymore and its solved, so we don't need [9599964](https://github.com/scaffold-eth/scaffold-eth-2/pull/963/commits/95999648a2f0cbd47c9e0ad0cef2137ad079277c)? 

> I met https://github.com/scaffold-eth/scaffold-eth-2/issues/901 again. I changed prettier file to .prettierrc.js and https://github.com/trivago/prettier-plugin-sort-imports/issues/131#issuecomment-1230926769, because of @trivago/prettier-plugin-sort-imports causes bug with underlines in vscode. It's a pity that I couldn't find it earlier, maybe https://github.com/scaffold-eth/scaffold-eth-2/issues/901 wont even started 

Ohh so #901 appeared again? Should we ask Matt to test again? Just in case because its #901 never happened to me so can't test :/

**Comment by rin-st** [comment]
> Umm I didn't get this changes 😅, also maybe its bad to pass it whole dir? and we should keep its as it was earlier what next lint by defults lints? Since next team might have researched to put all necessary to lint and ignore unnecessary stuff?

Without that change `next lint` doesn't find lint errors outside of that folders, for example in `tailwind.config.js`. So without linting all the nextjs folder I can't know if there's some lint error outside default folders. Same for `gh-actions` and `lint-staged` since they use `next:lint` and errors are not found. That's why after my first commits, error in `tailwind.config.js` was still existed

After [9599964](https://github.com/scaffold-eth/scaffold-eth-2/pull/963/commits/95999648a2f0cbd47c9e0ad0cef2137ad079277c) error is found by `next:lint` and hence by `gh-actions`. Also [b0fb91b](https://github.com/scaffold-eth/scaffold-eth-2/pull/963/commits/b0fb91bca2fdc9a1ed23767a24127b23fc29a558) `lintstagedrc`, so js files are also listed

Am I missing something?

> Ohh so https://github.com/scaffold-eth/scaffold-eth-2/issues/901 appeared again? Should we ask Matt to test again? Just in case because its https://github.com/scaffold-eth/scaffold-eth-2/issues/901 never happened to me so can't test :/

Yes, but it's fixed now.

@MattPereira hey, could you please test if https://github.com/scaffold-eth/scaffold-eth-2/issues/901 exists for you in  this branch? (Checkout to branch -> yarn install -> (probably restart vscode) -> test code snippet from mentioned pr

**Comment by technophile-04** [comment]
> If not, we can use just default next lint but I'm voting for linting everything

Umm I would vote for going with `next lint` default 😅 don't want to opinionated on it. 

Also I am not sure about linitng JS files at gh actions / linstaged https://github.com/scaffold-eth/scaffold-eth-2/pull/963/commits/b0fb91bca2fdc9a1ed23767a24127b23fc29a558 because people who are using JS believes in living on the edge? (move fast, break stuff is fine for them?), I think we are adding friction. Also I am not sure if people are actually using JS or not with SE-2 so lol I don't have strong opinion on this but slightly leaning towards not having lint checks and keep things as previous version

**Comment by rin-st** [comment]
> Umm I would vote for going with `next lint` default 😅 don't want to opinionated on it.
> 
> Also I am not sure about linitng JS files at gh actions / linstaged [b0fb91b](https://github.com/scaffold-eth/scaffold-eth-2/commit/b0fb91bca2fdc9a1ed23767a24127b23fc29a558) because people who are using JS believes in living on the edge? (move fast, break stuff is fine for them?), I think we are adding friction. Also I am not sure if people are actually using JS or not with SE-2 so lol I don't have strong opinion on this but slightly leaning towards not having lint checks and keep things as previous version

Ok, I reverted these changes, but we should remember then that linting works only for default `next lint` folders

**Comment by technophile-04** [comment]
Tysm Rinat!! Just texted Port to do final testing on this https://github.com/scaffold-eth/scaffold-eth-2/issues/901#issuecomment-2272401054

**Review by technophile-04 — APPROVED** [review]
Thanks @portdeveloper for test!!

**Pull request merged**